gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/pages/ErrorPage.java

```diff
@@ -43,14 +46,20 @@ public ErrorPage(final Exception e) {
 		// generate an error code so we can log the exception with it without giving the user the stacktrace
 		// note that wicket will already have logged the stacktrace so we aren't going to bother logging it again
 		final String code = RandomStringUtils.randomAlphanumeric(10);
-		log.error("User supplied error code for the above stacktrace: " + code);
+		log.error("User supplied error code for the above stacktrace: {}", code);
 
 		final Label error = new Label("error", new StringResourceModel("errorpage.text", null, new Object[] { code }));
 		error.setEscapeModelStrings(false);
 		add(error);
 
-		// show the stacktrace. This should be configurable at some point
-		add(new Label("stacktrace", stacktrace));
-
+		// Display the stack trace only if the application is configured to do so
+		WebMarkupContainer container = new WebMarkupContainer("stacktraceContainer");
+		Label trace = new Label("stacktrace", stacktrace);
+		if (!IExceptionSettings.SHOW_EXCEPTION_PAGE.equals(displayExceptionSetting) && !businessService.isSuperUser()) {
```

Comment:
You can eliminate all of this IExceptionSettings logic and just use the property directly in this page, rather than using the property to set another property.

Reply:
use wicket enclosure instead of a second container